hudi-spark/src/main/java/org/apache/hudi/OverwriteWithLatestAvroPayload.java

```diff
@@ -60,7 +60,18 @@ public OverwriteWithLatestAvroPayload preCombine(OverwriteWithLatestAvroPayload
   @Override
   public Option<IndexedRecord> combineAndGetUpdateValue(IndexedRecord currentValue, Schema schema) throws IOException {
     // combining strategy here trivially ignores currentValue on disk and writes this record
-    return getInsertValue(schema);
+    Object deleteMarker = null;
+    try {
+      deleteMarker = DataSourceUtils.getNestedFieldVal(
+          genericRecord, "_hoodie_delete_marker");
+    } catch (HoodieException e) {
```

Comment:
is there a better way to detect not found than throwing an exception.. this happens in the fast path, so may not be ideal to do it, this way

Reply:
I have introduced a new method in DataSourceUtils to return null if not found. Hope thats good enough